pkg/devfile/parser/parse.go

```diff
@@ -181,3 +178,34 @@ func parseParentAndPlugin(d DevfileObj) (err error) {
 
 	return nil
 }
+
+func parseFromURI(uri string, curDevfile DevfileObj) (DevfileObj, error) {
```

Comment:
looking at the tests, i would just pass the devfile context here rather than the entire devfile object

Reply:
good point! updated